src/headerssync.cpp

```diff
@@ -64,7 +64,8 @@ HeadersSyncState::HeadersSyncState(NodeId id, const Consensus::Params& consensus
     // exceeds this bound, because it's not possible for a consensus-valid
     // chain to be longer than this (at the current time -- in the future we
     // could try again, if necessary, to sync a longer chain).
-    m_max_commitments = 6*(Ticks<std::chrono::seconds>(NodeClock::now() - NodeSeconds{std::chrono::seconds{chain_start->GetMedianTimePast()}}) + MAX_FUTURE_BLOCK_TIME) / HEADER_COMMITMENT_PERIOD;
+    // Subtract cached headers as we don't redownload and verify commitments for them.
```

Comment:
aren't we subtracting the max possible headers (for some reason I don't yet understand)?

Reply:
Q: I don't yet understand how it can that be negative (that's why we need the `std::max` now, right)? But I'll debug it, you don't have to explain it, just documenting my surprises. Breaking it up into smaller named symbols could help.